### PR TITLE
feat: Create x/curation module data model and tx API

### PR DESCRIPTION
**Pull request by aaronc**
ref: #374

I had some ideas in my head on this so wanted to get them down. This proposes an API for the `x/curation` module as proposed in #374.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by ruhatch — COMMENTED** [review]
Looks like a reasonable API to me. 

Do we want to be able to delete a tagset as well?

Also a quick question on naming - some places the messages are name with `Request` and some places they're not. Do we have a consistent guideline for that?

**Comment by aaronc** [comment]
> Do we want to be able to delete a tag/tagset as well?

Likely not, I think.

> Also a quick question on naming - some places the messages are name with `Request` and some places they're not. Do we have a consistent guideline for that?

Remove `Request` is now the convention.

File: proto/regen/curation/v1/tx.proto [hunk 1]
```diff
@@ -0,0 +1,132 @@
+syntax = "proto3";
+package regen.curation.v1;
+
+option go_package = "github.com/regen-network/regen-ledger/x/curation";
+
+// Msg is the regen.curation.v1 Msg service.
+service Msg {
+    // CreateTagSet creates a new tag set with the provided metadata and returns an auto-generated tag set ID.
+    rpc CreateTagSet(MsgCreateTagSet) returns (MsgCreateTagSetResponse) {}
```
Comment (cyberbono3) [inline proto/regen/curation/v1/tx.proto:9]: I would rename `MsgCreateTagSetRequest` to `MsgCreateTagSet`. It applies to other rpc methods as well., e.g. `MsgCreateTagRequest` to `MsgCreateTag`...etc

Comment (blushi) [reply proto/regen/curation/v1/tx.proto:9]: see https://github.com/regen-network/regen-ledger/pull/395#issuecomment-870608008

File: proto/regen/curation/v1/tx.proto [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+    // denom_prefix is the prefix of denom's in this asset class. This could correspond to a credit class prefix which
+    // is always the prefix of all credit batches issued within this class. This provides a way to tag whole classes of
+    // credits or NFTs.
+    string denom_prefix = 3;
+}
+
+message MsgTagAssetClassResponse { }
+
+message MsgEditTagSetMetadata {
```
Comment (blushi) [inline proto/regen/curation/v1/tx.proto:106]: what if we only want to edit one field (eg name)? Should we just have the rest of the fields set with their current values?

**Comment by aaronc** [comment]
Closed in favor of #609

**Pull request closed without merging**